include/QBDI/VM.h

```diff
@@ -39,16 +40,16 @@ class InstrRule;
 struct MemCBInfo;
 
 class QBDI_EXPORT VM {
-    private:
+private:
     // Private internal engine
-    Engine*     engine;
+    std::unique_ptr<Engine> engine;
     uint8_t  memoryLoggingLevel;
-    std::vector<std::pair<uint32_t, MemCBInfo>>* memCBInfos;
+    std::unique_ptr<std::vector<std::pair<uint32_t, MemCBInfo>>> memCBInfos;
```

Comment:
``std::vector<std::pair<uint32_t, MemCBInfo>>`` a vector of std::pair could be rethought in my opinion

Reply:
Like engine, the used of a pointer allows to not define ``MemCBInfo`` in the public headers.

The structure is small and doesn't have any llvm element. We may define it in the private section of the VM.